pygmt/src/plot.py

```diff
@@ -207,6 +207,12 @@ def plot(self, x=None, y=None, data=None, size=None, direction=None, **kwargs):
     extra_arrays = []
     if "S" in kwargs and kwargs["S"][0] in "vV" and direction is not None:
         extra_arrays.extend(direction)
+    elif (
```

Comment:
It would be better if you can add a short comment here to explain what these lines does.